lib/update/cluster/phases/init.go

```diff
@@ -511,11 +536,15 @@ func removeLegacyUpdateDirectory(log log.FieldLogger) error {
 }
 
 // Rollback rolls back the init phase
-func (p *updatePhaseInit) Rollback(context.Context) error {
-	if err := p.removeConfiguredPackages(); err != nil {
+func (p *updatePhaseInit) Rollback(ctx context.Context) error {
+	if err := p.revertClusterConfig(); err != nil {
 		return trace.Wrap(err)
 	}
-	return nil
+
+	b := utils.NewExponentialBackOff(defaults.ShutdownTimeout)
+	return utils.RetryTransient(ctx, b, func() error {
+		return p.removeConfiguredPackages()
+	})
```

Comment:
This is just opportunistic improvement, right?  Or does it somehow relate to the main changes?

Reply:
Sorry, yes - an opportunistic improvement since it fails fairly consistently on rollback.